### PR TITLE
New makeProxy factory function in Swift

### PR DESCRIPTION
**Pull request by bernardnormier**
This PR adds a new `makeProxy` factory function to Ice for Swift.

It's similar to the existing uncheckedCast and should be preferred over uncheckedCast (and checkedCast) in many situations, in particular in the demos.

It also updated 3 tests to use makeProxy.

### CODE REVIEW

**Review by externl — APPROVED** [review]
Looks good!

File: swift/test/Ice/location/AllTests.swift [hunk 1]
```diff
@@ -13,14 +13,14 @@ func allTests(_ helper: TestHelper) throws {
...
+        communicator: communicator,
+        proxyString: "ServerManager:\(helper.getTestEndpoint(num: 0))",
         type: ServerManagerPrx.self
-    )!
+    )
 
     let locator = uncheckedCast(prx: communicator.getDefaultLocator()!, type: TestLocatorPrx.self)
-
-    let registry = try checkedCast(prx: locator.getRegistry()!, type: TestLocatorRegistryPrx.self)!
+    let registry = try uncheckedCast(prx: locator.getRegistry()!, type: TestLocatorRegistryPrx.self)
```
Comment (pepone) [inline swift/test/Ice/location/AllTests.swift:23]: In C++ we got rid of `uncheckedCast` usage with the constructor, will we do the same with makeProxy?

**Review by pepone — APPROVED** [review]
Looks good, I rerun Windows build but still failing. It works locally. Anyway is clearly unrelated to the PR.

File: swift/test/Ice/binding/AllTests.swift [hunk 2]
```diff
@@ -531,16 +521,14 @@ func allTests(_ helper: TestHelper) throws {
...
-            type: TestIntfPrx.self)
+        obj = obj.ice_endpointSelection(.Ordered)
         try test(obj.ice_getEndpointSelection() == .Ordered)
-        obj = uncheckedCast(prx: obj.ice_connectionCached(false), type: TestIntfPrx.self)
+        obj = obj.ice_connectionCached(false)
         try test(!obj.ice_isConnectionCached())
         let nRetry = 3
         var i = 0
         //
-        // Ensure that endpoints are tried in order by deactiving the adapters
```
[the remark this answers is not in the document]
Comment (bernardnormier) [reply swift/test/Ice/binding/AllTests.swift:543]: Fixed everywhere.

File: swift/test/Ice/ami/AllTests.swift [hunk 3]
```diff
@@ -13,13 +13,10 @@ func allTests(_ helper: TestHelper, collocated: Bool = false) throws {
     let output = helper.getWriter()
 
     var sref = "test:\(helper.getTestEndpoint(num: 0))"
```
[the remark this answers is not in the document]
Comment (bernardnormier) [reply swift/test/Ice/ami/AllTests.swift:15]: Removed in this test.

**Pull request merged**